lib/calc_route.dart

```diff
@@ -0,0 +1,138 @@
+import 'dart:math';
+import 'package:args/args.dart';
+import 'cart_point.dart';
+import 'route_model.dart';
+
+import 'package:sail_routing_dart/route_writer.dart';
+
+bool v_flag = false;
+
+void main(List<String> args) {
+  var parser = new ArgParser();
+  parser.addFlag("verbose", abbr: 'v', defaultsTo: false);
+  var results = parser.parse(args);
+  v_flag = results['verbose'];
+  Car$1$2oint
+
+  CartPoint start = new CartPoint(0.0, 0.0);
+  CartPoint end = new CartPoint(1.0, -10.2);
+  double wind = 3 * pi / 2;
+  RouteModel route = new RouteModel(start: start, end: end, wind_radians: wind);
+
+  calculateOptimal(route);
+  print(route);
+
+  RouteWriter rw = new RouteWriter();
+  rw.writeToFileFromRouteModel(route: route);
+  rw.run_python_plotter();
+}
+
+double euc_dist(CartPoint a, CartPoint b) {
+  return a.toPoint().distanceTo(b.toPoint());
+}
+
+void verbosePrint(dynamic s) {
+  if (v_flag) {
+    print(s);
+  }
+}
+
+CartPoint find_intersection(CartPoint a_point, CartPoint b_point, CartPoint c_point, CartPoint d_point) {
```

Comment:
We should make this a vector2 function, shouldn't be much of a change from what you have already though. Also, this should be a part of a shared lib file. I added the dart file. We could talk more about it tomorrow